lib/chef/http.rb

```diff
@@ -468,12 +468,12 @@ def version_retries
 
     # @api private
     def http_retry_delay
-      config[:http_retry_delay]
+      options[:http_retry_delay] ||= config[:http_retry_delay]
     end
 
     # @api private
     def http_retry_count
-      config[:http_retry_count]
+      options[:http_retry_count] ||= config[:http_retry_count]
```

Comment:
these lines to setup the default should really go into the initializer.

Reply:
Updated it.